lab5.py

```diff
@@ -0,0 +1,138 @@
+
+    def get_GDP(self):
+        return f'GDP: {self.__GDP}'
+    def get_government(self):
+        return f'Government:{self.__government}'
+
+    def calculate_population_density(self):
+        """
+        Calculate the population destiny of the country
+        """
+        return self.__population/self.__area
+
+    def __str__(self):
+        return f'{self.get_name()}, {self.get_capital()}, {self.get_code()},
+        {self.get_population()}, {self.get_area()}, {self.get_GDP()}, {self.get_government()}'
+
+    def __repr__(self):
+        return f'Country({self.__name!r}, {self.__capital!r}, {self.__code!r},
+        {self.__population!r}, {self.__area!r}, {self.__GDP!r}, {self.__government!r})'
+
+    def __del__(self):
+        """
+        Destructor for the Country class.
+        """
+        print("Called destructor")
+
+class Land:
+    """
+    Module containing the class Land
+    """
+    def __init__(self, name: str = ""):
+        self.name = name
+        self.countries = []
+
+    def add_country(self, country):
+        """
+        Add a country to the list of countries in the Land.
+        """
+        self.countries.append(country)
+
+    def calculate_population_density(self):
+        """
+        Calculate the population destiny of all countries in the Land.
+        """
+        return sum(country.calculate_population_density() for country in self.countries)
+    def sort_countries_by_GDP(self):
+        """
+        Sort the countries in the Land by gdp in descending order
+        :return: A list of countries sorted by gdp
+        """
+        return sorted(self.countries, key=lambda x: x._Country__GDP, reverse=True)
+
+    def print_top_countries_by_GDP(self, top_count):
+        """
+        Print the top countries in the Land by gdp
+        """
+        sorted_countries = self.sort_countries_by_GDP()
+        print(f'Top {top_count} countries by GDP:')
+        for i, country in enumerate(sorted_countries[:top_count]):
+            print(f'{i + 1}. {country.name} - GDP: {country._Country__GDP}')
+
+    def choose_country(self, min_GDP, preferred_government):
+        """
+        Choose a country in the Land based on min_gdp and preferred government
+        :return: The chosen country None if none are found
+        """
+        eligible_countries = [country for country in self.countries if country._Country__GDP >= min_GDP and country.government == preferred_government]
+
+        if not eligible_countries:
+            print("No eligible countries found.")
+            return None
+
+        return eligible_countries[0]
+
+def main(inst1, inst2, inst3):
+    for x in (inst1, inst2, inst3):
+        x.print_all()
+
+Country1 = Country("Ukraine", "Kyiv", +380, 43000000, 603628, 2000000, GovernmentType.REPUBLIC)
+Country2 = Country("Poland", "Warsaw", +48, 37000000, 322575, 150000000, GovernmentType.REPUBLIC)
+Country3 = Country("USA", "Washington", +1202, 331000000, 9834000, 0, GovernmentType.REPUBLIC)
```

Comment:
variable name should start from small letter and you can name them poland, use, ukraine